src/operator/tensor/la_op.cc

```diff
@@ -0,0 +1,399 @@
+.set_attr<nnvm::FInplaceOption>("FInplaceOption", [](const NodeAttrs& attrs)
+  { return std::vector<std::pair<int, int>>{{2, 0}}; })
+.set_attr<FCompute>("FCompute<cpu>", LaOpForward<cpu, 2, 2, 3, 1, gemm>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_la_gemm"})
+.add_arguments(LaMatrixMacParam::__FIELDS__())
+.add_argument("A", "NDArray-or-Symbol", "Tensor of input matrices")
+.add_argument("B", "NDArray-or-Symbol", "Tensor of input matrices")
+.add_argument("C", "NDArray-or-Symbol", "Tensor of input matrices");
+
+NNVM_REGISTER_OP(_backward_la_gemm)
+.set_num_inputs(4)
+.set_num_outputs(3)
+.set_attr_parser(ParamParser<LaMatrixMacParam>)
+.set_attr<nnvm::FInplaceOption>("FInplaceOption", [](const NodeAttrs& attrs)
+  { return std::vector<std::pair<int, int> >{{1, 0}, {2, 1}, {3, 2}}; })
+.set_attr<FResourceRequest>("FResourceRequest", [](const NodeAttrs& attrs)
+  { return std::vector<ResourceRequest>{ResourceRequest::kTempSpace}; })
+.set_attr<nnvm::TIsBackward>("TIsBackward", true)
+.set_attr<FCompute>("FCompute<cpu>", LaOpBackward<cpu, 2, 2, 4, 3, gemm_backward>);
+
+NNVM_REGISTER_OP(la_gemm2)
+.describe(R"code(Performs general matrix multiplication.
+Input are two tensors *A*, *B* each of dimension *n >= 2* and each
+having the same shape on the leading *n-2* dimensions. For every *n-2* dimensional index *i* let
+*A*\ :sub:`i`\ , *B*\ :sub:`i`\  be the matrices given by the last *2* dimensions.
+The operator performs the BLAS3 function *gemm* (restricted to two arguments)
+
+   *out*\ :sub:`i` = *alpha* \* *op*\ (*A*\ :sub:`i`\ ) \* *op*\ (*B*\ :sub:`i`\ )
+
+on all such pairs of matrices. Here *alpha* is a scalar operator parameter and *op()* is either
+the identity or the matrix transposition.
+
+In case of *n=2*, a single *gemm* function is performed on the matrices *A*, *B*.
+
+.. note:: The operator does only support float32 and float64 data types and provides
+          proper backward gradients.
+
+Examples::
+
+   // Single matrix multiply
+   A = [[1.0, 1.0], [1.0, 1.0]]
+   B = [[1.0, 1.0], [1.0, 1.0], [1.0, 1.0]]
+   la_gemm2(A, B, transpose_b = 1, alpha = 2.0)
+            = [[4.0, 4.0, 4.0], [4.0, 4.0, 4.0]]
+
+   // Batch matrix multiply
+   A = [[[1.0, 1.0]], [[0.1, 0.1]]]
+   B = [[[1.0, 1.0]], [[0.1, 0.1]]]
+   la_gemm2(A, B, transpose_b = 1, alpha = 2.0 )
+           = [[[4.0]], [[0.04 ]]]
+)code" ADD_FILELINE)
+.set_num_inputs(2)
+.set_num_outputs(1)
+.set_attr_parser(ParamParser<LaMatrixMultParam>)
+.set_attr<nnvm::FListInputNames>("FListInputNames", [](const NodeAttrs& attrs)
+  { return std::vector<std::string>{"A", "B"}; } )
+.set_attr<nnvm::FInferShape>("FInferShape", LaMatrixMultMacOpShape)
+.set_attr<nnvm::FInferType>("FInferType", ElemwiseType<2, 1>)
+.set_attr<FCompute>("FCompute<cpu>", LaOpForward<cpu, 2, 2, 2, 1, gemm2>)
+.set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_la_gemm2"})
+.add_arguments(LaMatrixMultParam::__FIELDS__())
```

Comment:
add_arguments(LaMatrixMultParam::__FIELDS__()) this should go after ndarray-or-symbol arguments